doc/man3/d2i_X509.pod

```diff
@@ -538,6 +540,22 @@ Alternative technique:
  if (d2i_X509(&x, &p, len) == NULL)
      /* error */
 
+Setting up a library context and property query:
+
+ X509 *x;
+ unsigned char *buf;
+ const unsigned char *p;
+ int len;
+ OSSL_LIB_CTX *libctx = ....;
+ const char *propq = ....;
+
+ /* Set up buf and len to point to the input buffer. */
+ p = buf;
+ x = X509_new_ex(libctx, propq);
+
+ if (d2i_X509(&x, &p, len) == NULL)
+     /* error, x was freed and NULL assigned to it */
```

Comment:
In that case, either in the prose documenting the *reuse* scenario, or more explicitly here, or both, add a short phrase to the effect that: "On error, the supplied pointer is freed via `OpenSSL_free`, which also means that it MUST be allocated via `OpenSSL_malloc` or similar".

Reply:
I just noticed:
```
d2i_TYPE(), d2i_TYPE_bio() and d2i_TYPE_fp() return a valid TYPE structure or NULL if an error occurs. If the "reuse" capability has been used with a valid structure being passed in via a, then the object is freed in the event of error and *a is set to NULL.
```
Under "RETURN VALUES". So perhaps just say "see RETURN VALUES", ...

Reply:
Done